File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def isPrime(number):
    if number == 1: return False
    if number == 2: return True
    for i in range(2, int(number**(0.5)) + 1):
        if number % i == 0:
            return False
    return True

def solution(n, k):
    answer = 0
    
    # k진수 변환하기
    number = ''
    while n:
        l, m = divmod(n, k)
        number += str(m)
        n = l
    number = number[::-1].split("0")
    
    for num in number:
        if num and isPrime(int(num)):
            answer += 1

    return answer

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
// 소수 판별 함수
const isPrime = (num) => {
    if (num === 1) return false;
    if (num === 2) return true;
    
    for (let i = 3; i <= parseInt(num**(1/2)); i++) {
        if (num % i === 0) return false;
    }
    
    return true;
}

function solution(n, k) {
    let answer = 0;
    
    const convertN = n.toString(k).split("0");
    for (const num of convertN) {
        if (num && isPrime(Number(num))) answer++;
    }
    
    return answer;
}